locksmith/src/operations/userMetadataOperations.ts

```diff
@@ -1,7 +1,16 @@
 import * as Normalizer from '../utils/normalizer'
 import { UserTokenMetadataInput } from '../types'
 import { UserTokenMetadata } from '../models'
+import { isEmpty } from 'lodash'
 
+export function isMetadataEmpty(data: Record<string, any>) {
+  const publicData = isEmpty(data?.public)
+  const protectedData = isEmpty(data?.protected)
+  return publicData && protectedData
+}
+
+// TODO: Are we still using this?
+// We should and we should add he logic of overriding data by merging
```

Comment:
@searchableguy I have found multiple places where we do `UserTokenMetadata.upsert` I think it would make sense to DRY things up, especially as we are starting to do some smarter things around conficts.

Reply:
Yeah, see the other PR I opened which does this.

Reply:
Cool! I'll wait for yours and rebase this one then